File: csharp/Services/PlayerServerService.cs
using Microsoft.Extensions.Logging;
using NewNanManager.Client.Http;
using NewNanManager.Client.Models;

namespace NewNanManager.Client.Services;

/// <summary>
/// 玩家服务器关系管理服务
/// </summary>
public class PlayerServerService : HttpClientBase
{
    public PlayerServerService(HttpClient httpClient, ILogger? logger = null)
        : base(httpClient, logger) { }

    /// <summary>
    /// 获取玩家的服务器关系
    /// </summary>
    /// <param name="playerId">玩家ID</param>
    /// <param name="onlineOnly">是否只显示在线服务器</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>玩家的服务器关系</returns>
    public async Task<PlayerServersData> GetPlayerServersAsync(
        int playerId,
        bool? onlineOnly = null,
        CancellationToken cancellationToken = default
    )
    {
        var queryParams = new Dictionary<string, object?> { ["online_only"] = onlineOnly };

        var queryString = BuildQueryString(queryParams);
        return await GetAsync<PlayerServersData>(
            $"/api/v1/players/{playerId}/servers{queryString}",
            cancellationToken
        );
    }

    /// <summary>
    /// 获取全局在线玩家
    /// </summary>
    /// <param name="page">页码</param>
    /// <param name="pageSize">每页大小</param>
    /// <param name="search">搜索玩家名</param>
    /// <param name="serverId">服务器ID（可选）</param>
    /// <param name="onlineOnly">是否只显示在线玩家</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>全局在线玩家列表</returns>
    public async Task<ServerPlayersData> GetServerPlayersAsync(
        int? page = null,
        int? pageSize = null,
        string? search = null,
        int? serverId = null,
        bool? onlineOnly = null,
        CancellationToken cancellationToken = default
    )
    {
        var queryParams = new Dictionary<string, object?>
        {
            ["page"] = page,
            ["page_size"] = pageSize,
            ["search"] = search,
            ["server_id"] = serverId,
            ["online_only"] = onlineOnly,
        };

        var queryString = BuildQueryString(queryParams);
        return await GetAsync<ServerPlayersData>(
            $"/api/v1/server-players{queryString}",
            cancellationToken
        );
    }

    /// <summary>
    /// 设置玩家离线状态 - 在玩家退出时调用
    /// </summary>
    /// <param name="serverId">服务器ID</param>
    /// <param name="playerIds">玩家ID列表</param>
    /// <param name="cancellationToken">取消令牌</param>
    /// <returns>空响应</returns>
    public async Task SetPlayersOfflineAsync(
        int serverId,
        int[] playerIds,
        CancellationToken cancellationToken = default
    )
    {
        var request = new { server_id = serverId, player_ids = playerIds };

        await PostAsync("/api/v1/servers/players/offline", request, cancellationToken);
    }
}

File: kotlin/src/main/kotlin/com/nanmanager/client/services/PlayerServerService.kt
package com.nanmanager.client.services

import com.nanmanager.client.models.*
import io.ktor.client.*

/**
 * 玩家服务器关系管理服务模块
 */
class PlayerServerService(
    httpClient: HttpClient,
    baseUrl: String,
    token: String
) : BaseService(httpClient, baseUrl, token) {



    /**
     * 获取玩家的服务器关系
     */
    suspend fun getPlayerServers(playerId: Int, onlineOnly: Boolean = false): PlayerServersResponse {
        val params = buildParams(mapOf("online_only" to onlineOnly))
        return get("/api/v1/players/$playerId/servers", params)
    }

    /**
     * 获取服务器的在线玩家
     */
    suspend fun getServerPlayers(request: GetServerPlayersRequest): ServerPlayersResponse {
        val params = buildParams(mapOf(
            "page" to request.page,
            "page_size" to request.pageSize
        ))
        return get("/api/v1/servers/${request.serverId}/players", params)
    }

    /**
     * 获取全局在线玩家
     */
    suspend fun getOnlinePlayers(request: GetOnlinePlayersRequest = GetOnlinePlayersRequest()): ServerPlayersResponse {
        val params = buildParams(mapOf(
            "page" to request.page,
            "page_size" to request.pageSize,
            "search" to request.search,
            "server_id" to request.serverId
        ))
        return get("/api/v1/online-players", params)
    }

    /**
     * 设置玩家离线状态 - 在玩家退出时调用
     */
    suspend fun setPlayersOffline(serverId: Int, playerIds: List<Int>): EmptyResponse {
        val request = mapOf(
            "server_id" to serverId,
            "player_ids" to playerIds
        )
        return post("/api/v1/servers/players/offline", request)
    }
}
